compiler/fir/resolve/src/org/jetbrains/kotlin/fir/resolve/transformers/body/resolve/FirExpressionsResolveTransformer.kt

```diff
@@ -187,9 +188,46 @@ class FirExpressionsResolveTransformer(transformer: FirBodyResolveTransformer) :
             }
 
         dataFlowAnalyzer.exitFunctionCall(completeInference, callCompleted)
+        if (callCompleted && completeInference.isArrayOfCall) {
+            return completeInference.toArrayOfCall().compose()
```

Comment:
What happens to calls of the form `arrayOf<Int>(1,2,3)`?

This is supposed to produce an array of boxed integers, but if I read the code correctly, `FirArrayOfCall` will produce primitive arrays depending on the element type and would result in an `IntArray` in this example, instead of an `Array<Int>`.

Reply:
Thank you for asking this question.

Note that `arrayOf` call is resolved, so is its return type. Once it's resolved as boxed array, like `Array<Int>`, somewhere below, `FirArrayOfCall`'s `typeRef` will be also same as boxed array. Then, during the conversion (from `fir` to `ir`), we use array type as-is, and retrieve element type from the array type, which would be boxed `Int`, not primitive `int`, since the given type is boxed array.